src/python/bot/fuzzers/syzkaller/fuzzer.py

```diff
@@ -0,0 +1,167 @@
+  """Return a suzkaller runner object."""
+  build_dir = environment.get_value('BUILD_DIR')
+  return SyzkallerRunner(fuzzer_path, build_dir)
+
+
+class Syzkaller(builtin.EngineFuzzer):
+  """Builtin syzkaller fuzzing engine."""
+
+  def generate_arguments(self, fuzzer_path):
+    """Generate arguments for fuzzer using .options file or default values."""
+    return ' '.join(get_arguments(fuzzer_path))
+
+
+class SyzkallerRunner(new_process.ProcessRunner):
+  """Syzkaller runner."""
+
+  def __init__(self, executable_path, default_args=None):
+    """Inits the SyzkallerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    super(SyzkallerRunner, self).__init__(
+        executable_path=executable_path, default_args=None)
+
+  def get_command(self, additional_args=None):
+    """Process.get_command override."""
+    base_command = super(SyzkallerRunner,
+                         self).get_command(additional_args=additional_args)
+
+    return base_command
+
+  def _create_empty_testcase_file(self):
+    """Create an empty testcase file in temporary directory."""
+    _, path = tempfile.mkstemp(dir=fuzzer_utils.get_temp_dir())
+    return path
+
+  def get_testcase_path(self, log_lines):
+    """Get testcase path from log lines."""
+    for line in log_lines:
+      match = re.match(KASAN_CRASH_TESTCASE_REGEX, line)
+      if match:
+        return match.group(1)
+
+    return None
+
+  def fix_timeout_argument_for_reproduction(self, arguments):
+    """Changes timeout argument for reproduction. This is higher than default to
+    avoid noise with smaller fuzzing defaults."""
+    fuzzer_utils.extract_argument(arguments, constants.TIMEOUT_FLAG)
+    arguments.append(
+        '%s%d' % (constants.TIMEOUT_FLAG, constants.REPRODUCTION_TIMEOUT_LIMIT))
+
+  def fuzz(self,
+           corpus_directories,
+           fuzz_timeout,
+           artifact_prefix=None,
+           additional_args=None,
+           extra_env=None):
+    """This is where actual syzkaller fuzzing is done."""
+    del corpus_directories, artifact_prefix, extra_env
+    additional_args = copy.copy(additional_args)
+    fuzz_result = self.run_and_wait(
+        additional_args=additional_args, timeout=fuzz_timeout)
+
+    log_lines = utils.decode_to_unicode(fuzz_result.output).splitlines()
+    fuzz_result.output = None
+    crash_testcase_file_path = self.get_testcase_path(log_lines)
+
+    if not crash_testcase_file_path and fuzz_result.return_code:
+      crash_testcase_file_path = self._create_empty_testcase_file()
+
+    fuzz_logs = '\n'.join(log_lines)
+
+    # TODO: Parse stats information and add them to FuzzResult
+    parsed_stats = []
+
+    crashes = []
+    if crash_testcase_file_path:
+      arguments = additional_args[:]
+      #libfuzzer.remove_fuzzing_arguments(arguments)
+      arguments = additional_args[:]
+      reproduce_arguments = arguments[:]
```

Comment:
there is a lot of duplication here.

Just do 

`reproduce_arguments = additional_args[:]`

That said, what do arguments look like here? Can these be passed as-is to syzkaller to reproduce the same bug? That's the intention of this argument.

Reply:
Thanks for clarifying that. I will assign it to None and add a TODO here to handle the repro params.